src/Microsoft.IdentityModel.Tokens/Encryption/EcdhKeyExchangeProvider.cs

```diff
@@ -0,0 +1,275 @@
+        {
+            if (privateKey == null)
+                throw LogHelper.LogArgumentNullException(nameof(privateKey));
+            if (publicKey is null)
+                throw LogHelper.LogArgumentNullException(nameof(publicKey));
+
+            ValidateAlgAndEnc(alg, enc);
+
+            SetKeyDataLenAndEncryptionAlgorithm(alg, enc);
+            _ecParamsPrivate = privateKey.ECDsa.ExportParameters(true);
+            _ecParamsPublic = publicKey.ECDsa.ExportParameters(false);
+
+            ValidateCurves(nameof(privateKey), nameof(publicKey));
+
+            Initialize();
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="EcdhKeyExchangeProvider"/> used for CEKs
+        /// <param name="privateKey">The <see cref="ECDsaSecurityKey"/> that will be used for cryptographic operations and represents the private key.</param>
+        /// <param name="publicKey">The <see cref="JsonWebKey"/> that will be used for cryptographic operations and represents the public key.</param>
+        /// <param name="alg">alg header parameter value.</param>
+        /// <param name="enc">enc header parameter value.</param>
+        /// </summary>
+        public EcdhKeyExchangeProvider(ECDsaSecurityKey privateKey, JsonWebKey publicKey, string alg, string enc)
+        {
+            if (privateKey == null)
+                throw LogHelper.LogArgumentNullException(nameof(privateKey));
+            if (publicKey is null)
+                throw LogHelper.LogArgumentNullException(nameof(publicKey));
+
+            ValidateAlgAndEnc(alg, enc);
+
+            SetKeyDataLenAndEncryptionAlgorithm(alg, enc);
+            _ecParamsPrivate = privateKey.ECDsa.ExportParameters(true);
+            _ecParamsPublic = GetEcParamsFromJwk(publicKey);
+
+            ValidateCurves(nameof(privateKey), nameof(publicKey));
+
+            Initialize();
+        }
+
+        /// <summary>
+        /// Generates the KDF
+        /// </summary>
+        /// <param name="apu">Agreement PartyUInfo (optional). When used, the PartyVInfo value contains information about the producer,
+        /// represented as a base64url-encoded string.</param>
+        /// <param name="apv">Agreement PartyVInfo (optional). When used, the PartyUInfo value contains information about the recipient,
+        /// represented as a base64url-encoded string.</param>
+        /// <returns>Returns <see cref="SecurityKey"/> that represents the key generated</returns>
+        public SecurityKey GenerateKdf(string apu = null, string apv = null)
+        {
+            //The "apu" and "apv" values MUST be distinct when used (per rfc7518 section 4.6.2) https://datatracker.ietf.org/doc/html/rfc7518#section-4.6.2
+            if (!string.IsNullOrEmpty(apu)
+                && !string.IsNullOrEmpty(apv)
+                && apu.Equals(apv, StringComparison.InvariantCulture))
```

Comment:
We should not specify culture here, we want OrdinalCaseSensitive, which is the default.

Reply:
Fixed in a52ecf360f5d0463192c760898765e2a8542ee68